inc/providers/class-core-sitemaps-posts.php

```diff
@@ -133,7 +149,20 @@ public function max_num_pages( $post_type = '' ) {
 			$post_type = $this->get_queried_type();
 		}
 
-		$query = new WP_Query(
+		/**
+		 * Filters the query arguments for calculating the maximum number of pages.
+		 *
+		 * Allows modification of the "maximum number of pages" query arguments before querying.
+		 *
+		 * @see WP_Query for a full list of arguments
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param array  $args      An array of WP_Query arguments.
+		 * @param string $post_type The post type string.
+		 */
+		$args = apply_filters(
+			'core_sitemaps_posts_max_num_pages_query_args',
```

Comment:
See my other comment related to taxonomies, it would be much better developer experience if we had a single `core_sitemaps_posts_query_args` filter - we could accomplish this by introducing a `get_public_posts( $post_type )` method that returns the common "base" query args.

Reply:
Now that I think about that... what about removing the filter completely... the "max num" is already part of the `$args` array, so there is no need to filter it at all... and it is also compatible with the other filter...

Reply:
I think the query args filter is unnecessary. the query args should fetch only the post ids using "fields" and leave it to plugins to do the filter using the post ids. The plugins can include or exclude post ids based on user selection. A similar approach should be adopted to taxonomy query args. These are better approaches as the plugins should anyway have a list of Post IDs to Term IDs to do the necessary filter. 

https://github.com/GoogleChromeLabs/wp-sitemaps/issues/187

Reply:
@Rajesh-Techblissonline but some might want to change the query... that’s what the mr is all about!

Reply:
@pfefferle That is fine. May be plugin authors can just make the query to retrieve POST IDs alone. But what is passed to the filter core_sitemaps_posts_url_list towards the end is url_list and if plugin authors like to add additional post attributes like lastmod or change frequency, he will need Post IDs. So the filter core_sitemaps_post_attributes within the POST loop will be better for him to add those attributes. Currently, the function adds only loc post attribute.

A filter like this will be better:

foreach ( $posts as $post ) {
		    $post_sitemap_attributes = array( 'loc' => get_permalink( $post )) ;
        	$url_list[] = apply_filters( 'core_sitemaps_post_attributes', $post_sitemap_attributes, $post );
	    }

This will let plugin authors to add their. $post_sitemap_attributes.

Reply:
@pfefferle core_sitemaps_posts_url_list will just give plugin authors a list of URLs and he will have to then retrieve other post_sitemap_attributes using an array of URLs and it will be inefficient.  Allowing them a filter within the POST loop will let him fetch other attributes using the POST ID.

Reply:
That‘s a completely different use case. Please open a new issue for such an enhancement.

Reply:
@swissspidy sure. I will raise a separate issue for that.

When it comes to this issue, if we allow a filter to modify Query arguments, plugin developers should make sure that they either retrieve entire POST, which is inefficient,  or POST ID along with other POST attributes they may require. But retrieving those additional POST attributes may not be necessary as there is no way for them to use those attributes further in the current function. However, POST ID is a must as it is needed to construct the url_list[] array using POST attributes. But as it stands now, the plugin author cannot add any additional POST attributes at the individual POST (or POST ID) level within the For Loop.

But the advantage of allowing a filter for the WP_Query arguments is they can exclude POSTS which is the main use-case for this filter and also decide the Order or retrieval ( ASC or DESC ).

Reply:
@swissspidy should the `$args` array for `get_url_list` and `max_num_pages` be the same?

Reply:
Let's continue this discussion in the other thread above